aips/aip-hybrid-nft-standard.md

```diff
@@ -0,0 +1,352 @@
+
+### Out of scope
+
+We are not covering the specifics of liquidity pools, that should already be handled automatically by any existing providers.
+
+## High-level Overview
+
+From a high level, when a user deposits a set number of fungible assets into an account (let's say one whole asset),
+they will receive an NFT in their account. When a user drops below the set number of fungible assets, the NFT will be
+removed from their account. Additionally, NFTs can be transferred directly to other users, and the fungible asset will
+go with it.
+
+### Asset Transfer Mechanisms
+
+The standard supports two distinct transfer mechanisms:
+
+1. **Digital Asset (NFT) Transfers**
+   - Supports direct digital asset transfers and marketplace trading
+   - Automatically manages corresponding fungible token transfers
+   - Ensures maintenance of the required fungible token balance per NFT
+   - Maintains atomic execution of both NFT and fungible token transfers
+
+2. **Fungible Asset Transfers**
+   - Executed through DEX trading or direct transfers
+   - Automatically manages NFT minting and burning based on token threshold
+   - When an account accumulates sufficient fungible tokens to constitute a "full NFT," an NFT is minted to that account
+   - When an account's balance falls below the "full NFT" threshold, the corresponding NFT is burned
+   - DEX implementations should include configurable NFT minting controls to optimize trading efficiency
+
+### Reveal Mechanics
+
+The standard provides flexible reveal mechanisms that collection creators can customize according to their requirements, allowing for diverse implementation strategies.
+
+## Impact
+
+NFT marketplaces and wallets will need to support the direct transfer of NFTs with fungible assets. This will allow them
+to ensure the properties above.
+
+## Alternative Solutions
+
+The alternative solution would be to have a non-standard way of creating these collections, which would cause
+fragmentation in nft marketplace, decentralized exchange, and wallet experiences.
+
+## Specification and Implementation Details
+
+### Technical Architecture
+The standard utilizes a resource account deployment model to enable universal access while maintaining security. The collection controller maintains mint/burn capabilities and ensures proper asset ratio maintenance. Key components include:
+
+- **Collection Controller**: Unique object managing collection operations
+- **HybridConfig**: Configuration object storing collection parameters
+- **HybridOwnershipData**: Tracks NFT ownership and enables ratio maintenance
+- **RevealRef**: Controls NFT metadata revelation processes
+
+The standard will be implemented through a factory contract deployed at:
+`0xbbe8a08f3b9774fccb31e02def5a79f1b7270b2a1cb9ffdc05b2622813298f2a`
+
+This factory contract will be deployed to a resource account, specifically to allow anyone to
```

Comment:
Might not be as important - it's better imo to use objects than resource accounts